packages/thirdweb/src/react/native/ui/connect/ConnectModal.tsx

```diff
@@ -0,0 +1,308 @@
+import { useState } from "react";
+import { Platform, StyleSheet, TouchableOpacity, View } from "react-native";
+import { SvgXml } from "react-native-svg";
+import type { Wallet } from "../../../../wallets/interfaces/wallet.js";
+import { parseTheme } from "../../../core/design-system/CustomThemeProvider.js";
+import type { Theme } from "../../../core/design-system/index.js";
+import type { ConnectButtonProps } from "../../../core/hooks/connection/ConnectButtonProps.js";
+import type { ConnectEmbedProps } from "../../../core/hooks/connection/ConnectEmbedProps.js";
+import { radius, spacing } from "../../design-system/index.js";
+import { useActiveWallet } from "../../hooks/wallets/useActiveWallet.js";
+import { useConnect } from "../../hooks/wallets/useConnect.js";
+import { getDefaultWallets } from "../../wallets/defaultWallets.js";
+import { ThemedButtonWithIcon } from "../components/button.js";
+import { Spacer } from "../components/spacer.js";
+import { ThemedText } from "../components/text.js";
+import { ThemedView } from "../components/view.js";
+import { BACK_ICON, CLOSE_ICON, TW_ICON, WALLET_ICON } from "../icons/svgs.js";
+import type { ModalState } from "./ConnectButton.js";
+import { ExternalWalletsList } from "./ExternalWalletsList.js";
+import { InAppWalletUI, OtpLogin } from "./InAppWalletUI.js";
+
+export function ConnectEmbed(props: ConnectEmbedProps) {
+  const theme = parseTheme(props.theme);
+  const wallet = useActiveWallet();
+  const adaptedProps = {
+    ...props,
+    connectModal: { ...props },
+  } as ConnectButtonProps;
+  return wallet ? null : (
+    <ConnectModal {...adaptedProps} theme={theme} containerType="embed" />
+  );
+}
+
+export function ConnectModal(
+  props: ConnectButtonProps & {
+    theme: Theme;
+    onClose?: () => void;
+    containerType: "modal" | "embed";
+  },
+) {
+  const { theme, client, containerType, accountAbstraction, onConnect } = props;
+  const connectMutation = useConnect({
+    client,
+    accountAbstraction,
+    onConnect,
+  });
+  const wallets = props.wallets || getDefaultWallets(props);
+  const [modalState, setModalState] = useState<ModalState>({ screen: "base" });
+  const inAppWallet = wallets.find((wallet) => wallet.id === "inApp") as
```

Comment:
I know we don't support ecosystem wallets in RN yet but might be good to call this something like `socialLoginWallet` and also check if the wallet id is an ecosystem wallet so we don't have to rename things when they're added to RN

Reply:
will do a pass for ecosystem wallets after this merges